core/unix/os.c

```diff
@@ -7834,13 +7834,28 @@ mmap_check_for_module_overlap(app_pc base, size_t size, bool readable, uint64 in
                  * Note, if it is a header of a different module, then we'll not have
                  * an overlap, so we will not hit this case.
                  */
-                ASSERT_CURIOSITY(
-                    ma->start + ma->os_data.alignment ==
-                    base
-                        /* On Mac we walk the dyld module list before the
-                         * address space, so we often hit modules we already
-                         * know about. */
-                        IF_MACOS(|| !dynamo_initialized && ma->start == base));
+#ifdef AARCH64
+                /* For AArch64 systems with 64K pages it is possible for small modules
+                 * (less than 1 page of disk space) to have multiple RW LOADable data
+                 * segments. Since these segments are mapped from a single disk space they
+                 * will all have a elf_header satisfying the check above.
+                 */
+                if (ma->os_data.alignment == 65536) {
```

Comment:
This could happen with 16K pages, or technically 4K, or other sizes: I don't think we should hardcode a particular size.

Reply:
Sure, better not to hardcode since it could happen with other page sizes.